### PR TITLE
48 make app more portable with renv

### PR DESCRIPTION
**Pull request by Xyarz**
added the renv with the lock file as well as a small Rmd to explain the usage of renv.

### CODE REVIEW

**Review by Robert-Krajcik — CHANGES_REQUESTED** [review]
Hi Lars

First thing: edit the DESCRIPTION file and add a comma in the authors@R field after this line
`person("Munshi Imran", "Hossain", role = "ctb")`
This is causing a DESCRIPTION read failure while running devtools::check()

Next please run golem::document_and_reload()
to update `man/riskassessment-package.Rd`

Thirdly, please add an entry to NEWS.md
Thanks,
Robet

**Comment by Xyarz** [comment]
There seems to be an issue with having both renv and pak.
It seems as if they just recently [updated](https://github.com/r-lib/actions/blob/v2-branch/setup-r-dependencies/action.yaml) the github action workflow in order to manage that.

**Comment by Robert-Krajcik** [comment]
Hi @Xyarz 
I see the `devtools::check()` works now, but the github actions are all failing.
This needs to get resolved before I can approve this.

Also note that the renv.Rmd file is not rendering
image

One way to fix this:
```{r setup, echo= FALSE}
pkgload::load_all(quiet = TRUE, warn_conflicts = FALSE, attach_testthat = FALSE)
library(riskassessment)
```
Another might be to simply remove the `library()` function

**Comment by borgmaan** [comment]
> Hi @Xyarz,
> 
> @borgmaan has initiated `renv` and it's passing `devtools::check()` on #322. Do you want to try to resolve the issue (the renv/pak conflict) you referenced on this branch , or just proceed forward with Andrew's branch? Please take a look and let us know your thoughts.

I would propose these two be kept separate and work continue over here to properly set up `renv` there is some nuance to doing this right and my goal with what was done on [`ab-add-shinytest2`](https://github.com/pharmaR/risk_assessment/tree/ab-add-shinytest2) was to quickly allow others to reproduce and confirm it was not a package compatibility issue. I plan to delete that `renv` prior to merging.

**Comment by AARON-CLARK** [comment]
At 1/31 meetup, @Xyarz said he'll prioritize this PR this week and the only remaining issue is related to libcurl. Thanks! I think this will be important for our upcoming release.

**Comment by Robert-Krajcik** [comment]
I'm looking this over now @borgmaan, but would like another set of eyes on this.  Thanks.

File: renv.lock
Comment (borgmaan) [inline renv.lock:7]: I propose pointing to a stable snapshot (either from MRAN or public RSPM). This will improve stability and prevent developers from always being prompted to upgrade packages. 

See [here](https://github.com/RConsortium/submissions-pilot2/blob/main/renv.lock#L7) for an example of this approach being used in the R submissions working group.

File: vignettes/renv.Rmd
Comment (borgmaan) [inline vignettes/renv.Rmd:38]: It may be worth adding logic here around the dev team philosophy of upgrading or changing the existing `lockfile` once it is stable. 

My suggestion would be to never upgrade once stable unless there is a very compelling reason to do so (e.g. new feature we can't live without).

**Review by borgmaan — CHANGES_REQUESTED** [review]
@Xyarz -- there is still some work to be done to get the GHA workflows working as expected with `renv`. 

All workflows should be updated to use `r-lib/actions/setup-renv` in place of `r-lib/actions/setup-r-dependencies`. Some of the workflows are currently using both (duplicative) and others are only using `r-lib/actions/setup-r-dependencies`. 

`r-lib/actions/setup-r-dependencies` uses `pak` under the hood and does not respect the `renv` lockfile. It includes it's [own logic](https://github.com/r-lib/actions/blob/v2/setup-r-dependencies/action.yaml#L78-L) for dependency resolution and can install versions that are mismatched with what is included in `renv.lock` leading to unexpected behavior. 

I dropped a couple of other comments regarding the repo source and philosophy on changing the lockfile once stable that may warrant feedback from the broader team (@AARON-CLARK ).

**Review by Robert-Krajcik — CHANGES_REQUESTED** [review]
I admit I'm new to working with `renv`
But I am getting this note:
```{r, eval=FALSE}
The following package(s) are used in the project, but are not installed:

	checkhelper
	revdepcheck
	riskassessment

Consider installing these packages, and then using `renv::snapshot()`
to record these packages in the lockfile.
```
When I go to install `revdepcheck`
```{r, eval=FALSE}
> install.packages("revdepcheck")
Error: package 'revdepcheck' is not available
In addition: Warning messages:
1: package 'revdepcheck' is not available 
2: package 'revdepcheck' is not available from MRAN
```
Can you look into this?

File: renv.lock
Comment (Xyarz) [reply renv.lock]: good idea thanks!

File: vignettes/renv.Rmd
Comment (AARON-CLARK) [reply vignettes/renv.Rmd:38]: That sounds good to me. @Xyarz, could you add a sentence or two that summarizes what Andrew has stated above?

Comment (Xyarz) [reply vignettes/renv.Rmd:39]: added in [a5a96fa](https://github.com/pharmaR/riskassessment/pull/315/commits/a5a96fa7644de7e69f087394effc1ab045739b32)

**Review by AARON-CLARK — APPROVED** [review]
LGTM

**Pull request merged**
